french_law/ocaml/law_source/.ocamlformat_ignore

```diff
@@ -0,0 +1 @@
+*
```

Comment:
You did that to speed up the build, because we don't need to format generated files ?

Reply:
Ah indeed that's something I wanted to discuss with you!
It feels a bit strange to me to reformat just-generated files; that also adds a run-time dependency to `ocamlformat` for running these rules. And it complicated things (we don't want ocamlformat and dune to compete on the resulting file).

No big speedup here (but in the case of the Python files, `autopep8` is very slow)

But I guess there were reasons to do this in the first place so I am open to finding a workaround if needed!

Reply:
You need to at least read the input type declarations to know what to feed your program. And since the pretty-printing in the backends is not ideal we run ocamlformat afterwards. @EmileRolley even calls directly ocamlformat in his new `api_web` backend plugin.

Reply:
Indeed! Just discussed the matter quickly with @EmileRolley and we agreed to disable it and focus on improving our pretty-printer on cases where readability is bad (from a quick check, it seems o.k. on type definitions). Otherwise, our pretty-printer will be completely untested: if we don't want to bother, and use OCamlformat, I'd say go all the way, simplify our pretty-printer into a bare-printer and call `ocamlformat` as a lib.

What do you think ?

Reply:
I'm okay with relying on our pretty-printer and improving it as needed, dropping `ocamlformat` altogether. Using `ocamlformat` as a lib is tempting but would introduce a lot of complexity as we use only a small portion of the OCaml AST. A textual pretty-printer would be simpler for maintainers to come and just change a tidbit while not having to understand everything. Plus that way we don't have to keep in sync with `ocamlformat`'s AST changes, while OCaml textual syntax is somewhat backwards-compatible.